src/stream/manager.rs

```diff
@@ -136,6 +136,7 @@ pub fn remove_stream(stream_name: &str) -> Result<(), SimpleError> {
     let mut manager = MANAGER.as_ref().lock().unwrap();
     match manager.streams.iter().position(find_stream) {
         Some(index) => {
+            manager.streams[index].stream_type.mut_inner().stop();
```

Comment:
That's a bit weird, why the stream can't stop itself on the desconstructor ?

Reply:
How would the stream know that it needs to be deconstructed once the client has clicked to 'remove' the stream? That's why I needed to add this here, and to me, it seems coherent with the `start()` called in the `add_stream_and_start()`.

Reply:
When you remove something from the list, the item will not have a owner and will call the constructor, there is no need to call stop manually, it should be done in the desconstructor.

Reply:
Hmm, interesting design, I don't know why, but it is not how it is working. I'll try to debug it.

Reply:
Perfect, I'd forgotten about implementing Drop() to call stop().